### PR TITLE
DOCSP-45058-mongosh-undefined-behavior

### PR DESCRIPTION
**Pull request by amalhotra-mdb**
## DESCRIPTION
[DOCSP-44560](https://jira.mongodb.org/browse/DOCSP-44560) notes that `mongosh` converts undefined to null during write operations, unlike the legacy `mongo` shell; this is not properly documented in the mongosh docs, so this PR adds this info to the [Compatibility Changes w/ Legacy `mongo` shell page](https://www.mongodb.com/docs/mongodb-shell/reference/compatibility/#compatibility-changes-with-legacy-mongo-shell)

## STAGING
https://deploy-preview-364--docs-mongodb-shell.netlify.app/reference/compatibility/#undefined-values

## JIRA
https://jira.mongodb.org/browse/DOCSP-45058

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/reference/compatibility.txt
Comment (sarah-olson-mongodb) [inline source/reference/compatibility.txt:210]: (question)
Do you think we could replace "stores a null value" with something a little less jargony, that retains the same meaning? Maybe add something about the impact to to document or to data when undefined is converted to null?

File: source/reference/compatibility.txt
Comment (sarah-olson-mongodb) [inline source/reference/compatibility.txt:211]: (suggestion)
Maybe omit this last sentence because I don't think we want to point folks to the legacy shell. Is there maybe another solution?

File: source/reference/compatibility.txt
Comment (sarah-olson-mongodb) [inline source/reference/compatibility.txt:234]: (question) 
I'm curious about whether there's another workaround. It might be good to check with @addaleax during tech review.

**Review by sarah-olson-mongodb — COMMENTED** [review]
Thanks so much for this update @amalhotra-mdb! And fantastic job identifying this doc gap and creating a ticket to resolve it! This all looks really good. 

My main question is around whether we want to point folks to the legacy mongo shell as the workaround? I think you might want to check in with @addaleax or @GaurabAryal, if you haven't already, regarding whether there might be another workaround. 

I'll be OOO next week and you don't need to wait on me to get this internal review LGTM'd. :)

File: source/reference/compatibility.txt
Comment (amalhotra-mdb) [reply source/reference/compatibility.txt]: After a bit of investigation, it seems that generally users should never actually need to insert an undefined value, so I'm thinking we might be okay to not provide a work around? I will get more opinions on this though. It seems if we really want to point them at a workaround, they could use a driver that supports inserting raw BSON data, e.g. through a Python script

**Review by nvillahermosa-mdb — APPROVED** [review]
Short and clear, LGTM

File: source/reference/compatibility.txt
Comment (FGasper) [inline source/reference/compatibility.txt:212]: It‘s also relevant that mongosh will represent any stored BSON undefined values as null. So if you’ve stored a BSON undefined value with some other tool—e.g., Go’s driver or the `mongo` shell—then you fetch that document with mongosh, you’ll see null where the actual stored document has undefined.

Should this section mention that as well?

File: source/reference/compatibility.txt
Comment (FGasper) [inline source/reference/compatibility.txt:209]: BSON undefined is deprecated everywhere, not just in mongosh.

**Pull request merged**
